Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabPublicHandler.cpp

```diff
@@ -1031,6 +1031,20 @@ namespace AzToolsFramework
             return path;
         }
 
+        TemplateId PrefabPublicHandler::GetOwningInstancePrefabTemplateId(AZ::EntityId entityId) const
+        {
+            InstanceOptionalReference instance = GetOwnerInstanceByEntityId(entityId);
+
+            if (instance.has_value())
```

Comment:
nit: this can be condensed into the if statement as follows:

```
if(InstanceOptionalReference instance = GetOwnerInstanceByEntityId(entityId); instance.has_value())
{
    return instance->get().GetTemplateId();
}

return Prefab::InvalidTemplateId;
```